core/templates/services/local-storage.service.ts

```diff
@@ -195,6 +199,125 @@ export class LocalStorageService {
     }
     return null;
   }
+
+  /**
+   * Retrieves the entity editor browser tabs info stored in the local storage
+   * for opened tabs of a particular entity type depending upon the
+   * browser tabs info constant.
+   * @param entityEditorBrowserTabsInfoConstant The key of the data stored on
+   * the local storage. It determines the type of the entity
+   * (topic, story, exploration, skill) for which we are retrieving data.
+   * @param entityId The id of the entity.
+   * @returns EntityEditorBrowserTabsInfo if the data is found on the local
+   * storage. Otherwise, returns null.
+   */
+  getEntityEditorBrowserTabsInfo(
+      entityEditorBrowserTabsInfoConstant: string, entityId: string
+  ): EntityEditorBrowserTabsInfo | null {
+    if (this.isStorageAvailable()) {
+      const allEntityEditorBrowserTabsInfoObjects:
+        EntityEditorBrowserTabsInfoObject[] = JSON.parse((
+        this.storage as Storage).getItem(
+          entityEditorBrowserTabsInfoConstant) || '[]');
```

Comment:
In what cases would the '[]' arise?

Also, don't have a default string and then JSON.parse it. Just assign the desired value as a default outside JSON.parse.

Reply:
Done. It would arise when initially, no tab of that particular entity type was opened and we are opening the first tab of this type.